File: Trade_project/app/routers/invoices.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app import models, schemas
from app.database import get_db

router = APIRouter()
invoices_router = APIRouter()

@invoices_router.post("/invoices/", response_model=schemas.Invoice)
def create_invoice(invoice: schemas.InvoiceCreate, db: Session = Depends(get_db)):
    db_invoice = models.Invoice(**invoice.dict())
    db.add(db_invoice)
    db.commit()
    db.refresh(db_invoice)
    return db_invoice

@invoices_router.get("/invoices/", response_model=list[schemas.Invoice])
def get_invoices(skip: int = 0, limit: int = 10, db: Session = Depends(get_db)):
    invoices = db.query(models.Invoice).offset(skip).limit(limit).all()
    return invoices

File: Trade_project_Frontend/src/api/invoices.js
import axiosInstance from './axiosInstance';

export const fetchInvoices = async () => {
    const response = await axiosInstance.get('/invoices/');
    return response.data;
};

export const createInvoice = async (invoice) => {
    const response = await axiosInstance.post('/invoices/', invoice);
    return response.data;
};
